Source/missiles.cpp

```diff
@@ -388,17 +388,23 @@ void CheckMissileCol(Missile &missile, int minDamage, int maxDamage, bool isDama
 	int mx = position.x;
 	int my = position.y;
 
+	MissileSource missileSource = missile.sourceType();
+
 	bool isMonsterHit = false;
 	int mid = dMonster[mx][my];
 	if (mid > 0 || (mid != 0 && Monsters[abs(mid) - 1].mode == MonsterMode::Petrified)) {
 		mid = abs(mid) - 1;
-		if (missile.IsTrap()
-		    || (missile._micaster == TARGET_PLAYERS
-		        && (Monsters[missile._misource].flags & MFLAG_TARGETS_MONSTER) != 0
-		        && ((Monsters[mid].flags & MFLAG_GOLEM) != 0 || (Monsters[missile._misource].flags & MFLAG_BERSERK) != 0))) {
-			isMonsterHit = MonsterTrapHit(mid, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted);
-		} else if (IsAnyOf(missile._micaster, TARGET_BOTH, TARGET_MONSTERS)) {
+		switch (missileSource) {
+		case MissileSource::Player:
 			isMonsterHit = MonsterMHit(missile._misource, mid, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted);
+			break;
+		case MissileSource::Monster:
+			if ((missile.sourceMonster()->flags & MFLAG_TARGETS_MONSTER) == 0 || (Monsters[mid].flags & (MFLAG_GOLEM | MFLAG_BERSERK)) == 0)

```

Comment:
previously this was checking if the attacking monster was beserked or the target monster is a golem, now it's checking if the target monster is beserked or a golem. Is that intentional?

Reply:
Is this function unit testable?

Reply:
It's not directly exposed, but you could create a test function that calls it. Similarly to how it was done for `GetSizeForThemeRoom` https://github.com/diasurgical/devilutionX/blob/master/Source/levels/gendung.cpp#L354

Reply:
I mean regarding the global variables, this function depends on many of those. I checked the other tests for examples of how to test this but didn't find any.

Reply:
Most can probably be ignored :)

Reply:
I tried to express this one a bit clearer in https://github.com/diasurgical/devilutionX/pull/5107, as far as I could tell the "does the target have the golem flag" check was serving double duty for detecting if a monster is berserked or controlled by a player. Seems like the correct approach was to check if either monster has the berserk flag.